src/systemdhook.c

```diff
@@ -825,6 +841,35 @@ int main(int argc, char *argv[])
 	yajl_val v_mount = yajl_tree_get(config_node, mount_label_path, yajl_t_string);
 	mount_label = v_mount ? YAJL_GET_STRING(v_mount) : "";
 
+	/* Get the gid value. */
+        const char *gid_mappings[] = {"linux", "gidMappings", (const char *)0 };
+	yajl_val v_gidMappings = yajl_tree_get(config_node, gid_mappings, yajl_t_array);
+	if (!v_gidMappings) {
+		pr_perror("gidMappings not found in config");
+		return EXIT_FAILURE;
+	}
+
+	yajl_val v_gidMappings_values = YAJL_GET_ARRAY(v_gidMappings)->values[0];
```

Comment:
We shouldn't assume that the first value is the correct one. We can calculate the host ID for container 0 by looking at hostID, containerID and Size.